h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -684,671 +493,39 @@ private int nextInstanceCounter(String algoName, NonBlockingHashMap<String, Inte
     }
   }
 
-  private Key<Model> modelKey(String algoName) {
-    return modelKey(algoName, true);
-  }
-
-  private Key<Model> modelKey(String algoName, boolean with_counter) {
+  Key<Model> modelKey(String algoName, boolean with_counter) {
     String counterStr = with_counter ? "_" + nextInstanceCounter(algoName, algoInstanceCounters) : "";
     return Key.make(algoName + counterStr + "_AutoML_" + timestampFormatForKeys.format(this.startTime));
   }
 
-  Job<Model> trainModel(Key<Model> key, WorkAllocations.Work work, Model.Parameters parms) {
-    return trainModel(key, work, parms, false);
-  }
-
```

Comment:
common `trainModel` logic moved with barely no changes to `TrainingStep.ModelStep` inner class.